src/compiler/checker.ts

```diff
@@ -3869,6 +3966,10 @@ namespace ts {
                 const signatureDeclaration = <SignatureDeclaration>node.parent;
                 const signature = getSignatureFromDeclaration(signatureDeclaration);
                 const parameterIndex = ts.indexOf(signatureDeclaration.parameters, node);
+                if (skipSignatureCheck) {

```

Comment:
Why is this check this far down and specific to nodes with initializers? (I'd expect it to be at least 3 lines up to avoid that work regardless).